server.go

```diff
@@ -114,10 +122,12 @@ func StartElection(s *Server) {
 	for _, val := range s.Servers {
 		// Let's assume he votes for himself
         if val.ID != s.ID {
-            //fmt.Printf("%v, calling %v at %v\n",s.ID, val.ID, val.Port)
+					if s.AliveServers[val.ID]{
+            fmt.Printf("%v alive servers\n", s.NumAliveServers)
             go RequestVote(s, val)
+					}
         }
-		if x := CheckVotes(s); x > numServers/2 {
+		if x := CheckVotes(s); x > s.NumAliveServers/2 {

```

Comment:
This isn't entirely correct. If we start with 5 servers, it should need 3 votes to get elected. It doesn't scale.